### DISCUSSION TITLE
Issues Compiling the Solution on VS 2019

### DISCUSSION THREAD

**Discussion by bcalco** [leppie/IronScheme#83] [Discussions]
@leppie  I embarked on compiling the solution from the github repository (technically, a fork of the github repository) and encountered some issues.

First off, I had to install .NET 3.1 Service Pack 1 before anything would compile. This in turn required activating the .NET 2.0 feature. Not a big deal.

Second, for some reason when rebuilding the whole solution in Release configuration, the IronScheme.FrameworkPAL project just hangs. Fortunately I previously compiled it successfully, so I have the binaries and it doesn't prevent the rest of the projects from compiling individually. I'll see tomorrow if the same behavior is exhibited after a fresh Repent, Reflect, Reboot cycle.

Third, at the command line in the `.\bin\[Debug|Release]\net20\` folders I cannot seem to run `(compile-system-libraries)`:

```
PS C:\projects\dotnet\IronScheme\IronScheme\IronScheme.Console\bin\Release\net20> .\IronScheme.Console.exe
IronScheme Latest github.com/IronScheme © 2007-2019 Llewellyn Pritchard (.NET 2.0 64-bit)
> (compile-system-libraries)
Unhandled exception during evaluation:
&error
&who: expander
&message: "cannot locate library in library-path"
&library-resolution:
  library: (srfi :0)
  files: ("C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20\\lib/srfi/%3a0.ironscheme.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20\\lib/srfi/%3a0.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20\\lib/srfi/%3a0.ss" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20/srfi/%3a0.ironscheme.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20/srfi/%3a0.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Release\\net20/srfi/%3a0.ss")

Statistics for 'total compile time':
  Real Time:  2123ms
  CPU Time:   2094ms
  User Time:  2156ms
  GC's:       99
> (exit)
PS C:\projects\dotnet\IronScheme\IronScheme\IronScheme.Console\bin\Release\net20> cd ..\..\Debug\net20\
PS C:\projects\dotnet\IronScheme\IronScheme\IronScheme.Console\bin\Debug\net20> .\IronScheme.Console.exe
IronScheme Latest github.com/IronScheme © 2007-2019 Llewellyn Pritchard (.NET 2.0 64-bit)
> (compile-system-libraries)
Unhandled exception during evaluation:
&error
&who: expander
&message: "cannot locate library in library-path"
&library-resolution:
  library: (srfi :0)
  files: ("C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20\\lib/srfi/%3a0.ironscheme.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20\\lib/srfi/%3a0.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20\\lib/srfi/%3a0.ss" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20/srfi/%3a0.ironscheme.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20/srfi/%3a0.sls" "C:\\projects\\dotnet\\IronScheme\\IronScheme\\IronScheme.Console\\bin\\Debug\\net20/srfi/%3a0.ss")

Statistics for 'total compile time':
  Real Time:  2289ms
  CPU Time:   2078ms
  User Time:  2141ms
  GC's:       100
>
```

I would really like to understand everything going on, all the deployment configuration requirements, etc. I expect to create additional system libraries specific to my application that I'd also like to boot. Also want to understand dynamic compilation and binary generation. My app will run on Windows (for now, till I develop and ASP.NET Core app that will run on Linux) so I can use the full framework, but when I do need to deploy some assets to Linux on .NET Core 5 (and eventually 6) I will also need to understand the various exigencies.

Now that I'm compiling locally, what's the best way to pull the dependencies into my project, including source libraries?

**Comment by leppie** [comment]
1. The project is quite old, sorry
2. .NET 3.5 SP1 should used I think as a minimum
3. Works fine for me. Possibly related to above

As for using IronScheme, I normally just create a program and libraries with a fresh extraction of the release zip file. I dont know what other people do. I used to have a Nuget for this, but I cant recall why it is not working or why I am not bothering with it anymore.

What you have set up is more of an environment for the compiler. Compiler changes are complicated. It might be useful for debugging though.

**Comment by bcalco** [comment]
What I want to know is how to (re)compile the boot file, possibly adding/removing things as I need to for my application. I'd like to understand what-all is in the `(interactive-environment)` and how to ... interact with it.

IOW, I also want to know how to access environment/state in a running program. Using the `string.Eval(...)` functions for a day or so not in earnest it's clear state is being maintained somewhere, somehow, but I don't really understand how to interact with it from within my program, at least not yet. So pointers on that would be appreciated.

Thanks for any help you can provide, @leppie - I appreciate it!
